docs/horizon.md

```diff
@@ -0,0 +1,71 @@
+# Horizon integration
+
+It is possible to upload images from the web UI provided by Horizon.
+On systems where the value of `LimitRequestBody` is not explicitly specified in
+an `httpd` configuration, the [default value](https://access.redhat.com/articles/6975397)
+is 1GiB.
+As a consequence, if the total size of the HTTP request body exceeds the 1
+GiB limit, httpd returns the `413 Request Entity Too Large` error code.
+While one possibility is to configure `LimitRequestBody` in Horizon httpd, the
+recommended option is to enable `CORS`.
+In general Horizon has different available options.
+
+1. `off`: disables the ability to upload images via Horizon.
+2. `legacy`: enables local file upload by piping the image file through the
+   Horizon’s web server.
+2. `direct` sends the image file directly from the web browser to Glance.
+
+"direct" is the preferred mode, and it requires `CORS` support to be enabled on
+the Glance API service.
+`direct` mode has several benefits:
```

Comment:
Perhaps it would be prudent to also mention the downside of the direct mode, that is that the Glance endpoint needs to be available outside for it to work.

Reply:
maybe we can clarify it, but in general if those are not available `Glance` won't reach a `Setup Complete` state.

Reply:
They could be blocked outside, though? I know that some customers only make the Horizon available to the outside, and not all of the OpenStack APIs. In that case the direct mode wouldn't work.

Reply:
Fair enough, but I don't think that the `Route` in k8s is blocked by default, and in order to do that, someone has to write a `NetworkPolicy` to explicitly don't make it reachable from a given `Pod`, which would make any additional note obvious to the human administrator because it might simply result in contradiction.

Reply:
If you think it's obvious, then fine.